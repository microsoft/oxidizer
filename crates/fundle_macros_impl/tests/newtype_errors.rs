// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

use syn::{parse_quote, ItemStruct, ItemUnion};

mod util;

#[test]
fn empty_structs() {
    let item: ItemStruct = parse_quote! {
        #[newtype]
        struct Foo;
    };

    insta::assert_snapshot!(expand_fundle_newtype!(item));
}

#[test]
fn multi_field() {
    let item: ItemStruct = parse_quote! {
        #[newtype]
        struct Foo(String, String);
    };

    insta::assert_snapshot!(expand_fundle_newtype!(item));
}

#[test]
fn multi_field_named() {
    let item: ItemStruct = parse_quote! {
        #[newtype]
        struct Foo {
            x: String,
            y: String
        }
    };

    insta::assert_snapshot!(expand_fundle_newtype!(item));
}


#[test]
fn union() {
    let item: ItemUnion = parse_quote! {
        #[newtype]
        union Foo {
            x: String,
            y: String
        }
    };

    insta::assert_snapshot!(expand_fundle_newtype!(item));
}