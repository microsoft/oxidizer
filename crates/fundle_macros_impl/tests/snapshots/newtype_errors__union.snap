---
source: crates/fundle_macros_impl/tests/newtype_errors.rs
assertion_line: 51
expression: expand_fundle_newtype!(item)
---
::core::compile_error! {
    "fundle::newtype can only be applied to structs"
}
